src/setup_payload/AdditionalDataPayloadGenerator.cpp

```diff
@@ -106,7 +106,7 @@ CHIP_ERROR AdditionalDataPayloadGenerator::generateRotatingDeviceId(uint16_t lif
     outputBufferWriter.Put(&hashOutputBuffer[kSHA256_Hash_Length - RotatingDeviceId::kRotatingDeviceIdHashSuffixLength],
                            RotatingDeviceId::kRotatingDeviceIdHashSuffixLength);
 
-    for (; rotatingDeviceIdBufferIndex < outputBufferWriter.Needed(); rotatingDeviceIdBufferIndex++)
+    for (rotatingDeviceIdBufferIndex = 0; rotatingDeviceIdBufferIndex < outputBufferWriter.Needed(); rotatingDeviceIdBufferIndex++)
```

Comment:
Please replace with function from BytesToHex.h at earliest convenience. Does not need to be in this PR, but it could be.

Reply:
Sure, i will do that in a following PR